kernel/src/sched.rs

```diff
@@ -989,15 +989,31 @@ impl Kernel {
             Syscall::Exit {
                 which,
                 completion_code,
-            } => match which {
-                // The process called the `exit-terminate` system call.
-                0 => process.terminate(completion_code as u32),
-                // The process called the `exit-restart` system call.
-                1 => process.try_restart(completion_code as u32),
-                // The process called an invalid variant of the Exit
-                // system call class.
-                _ => process.set_syscall_return_value(SyscallReturn::Failure(ErrorCode::NOSUPPORT)),
-            },
+            } => {
+                // Parse the `which` value according to the definition of the
+                // exit syscall.
+                let exit_type = match which {
+                    // The process called the `exit-terminate` system call.
+                    0 => Some(ExitCall::Terminate),
+                    // The process called the `exit-restart` system call.
+                    1 => Some(ExitCall::Restart),
+                    _ => None,
+                };
+
+                // Handle each variant of the exit syscall.
+                exit_type.map_or_else(
+                    || {
+                        // The process called an invalid variant of the Exit
+                        // system call class.
+                        process
+                            .set_syscall_return_value(SyscallReturn::Failure(ErrorCode::NOSUPPORT));
+                    },
+                    |exit| match exit {
+                        ExitCall::Terminate => process.terminate(completion_code as u32),
+                        ExitCall::Restart => process.try_restart(completion_code as u32),
+                    },
+                );
+            }
```

Comment:
```suggestion
            } => match which {
                // The process called the `exit-terminate` system call.
                ExitCall::Terminate as u32 => process.terminate(completion_code as u32),
                // The process called the `exit-restart` system call.
                ExitCall::Restart as u32 => process.try_restart(completion_code as u32),
                // The process called an invalid variant of the Exit
                // system call class.
                _ => process.set_syscall_return_value(SyscallReturn::Failure(ErrorCode::NOSUPPORT)),
            },
```

I think this is a less convoluted use, even if it doesn't enforce handling each variant of the exit system call class.

Reply:
```
   Compiling kernel v0.1.0 (/Users/bradjc/git/tock/kernel)
error: expected one of `!`, `(`, `...`, `..=`, `..`, `::`, `=>`, `if`, `{`, or `|`, found keyword `as`
   --> kernel/src/sched.rs:995:41
    |
995 |                     ExitCall::Terminate as u32 => process.terminate(completion_code as u32),
    |                                         ^^ expected one of 10 possible tokens

warning: unused import: `ExitCall`
  --> kernel/src/sched.rs:33:22
   |
33 | use crate::syscall::{ExitCall, Syscall, YieldCall};
   |                      ^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error: aborting due to previous error; 1 warning emitted

error: could not compile `kernel`
```